### PR TITLE
verify client gid uniqueness for a single service event.

### PR DESCRIPTION
**Pull request by fujitatomoya**
part of https://github.com/ros2/rmw/issues/357

- this PR makes sure uniqueness of client GID via single service event.
- at this moment, `rmw_connextdds` is the only rmw implementation that can pass this test.

### CODE REVIEW

File: rclcpp/test/rclcpp/test_service_introspection.cpp [hunk 1]
```diff
@@ -124,6 +124,20 @@ TEST_F(TestServiceIntrospection, service_introspection_nominal)
   ASSERT_THAT(
     client_gid_arr,
     testing::Eq(event_map[ServiceEventInfo::REQUEST_SENT]->info.client_gid));
+  // TODO(@fujitatomoya): Remove this if statement once other rmw implementations support test.
+  // Only rmw_connextdds can pass this test requirement for now.
+  // See more details for https://github.com/ros2/rmw/issues/357
+  if (std::string(rmw_get_implementation_identifier()).find("rmw_connextdds") == 0) {
+    ASSERT_THAT(
```
Comment (fujitatomoya) [inline rclcpp/test/rclcpp/test_service_introspection.cpp:131]: confirmed that `rmw_fastrtps` and `rmw_cyclonedds` fail in this ASSERTION.

**Comment by fujitatomoya** [comment]
@wjwwood @clalancette @ahcorde can you review this?

> It would be good to have a new test (maybe in rcl?) that checks that this condition is met, along with any clarification in the rmw docs to require it of the middleware.

https://github.com/ros2/rcl_interfaces/blob/f6b3741d921ef5414125091b2ccc74b863be3c24/service_msgs/msg/ServiceEventInfo.msg#L12-L17 and with this PR.

> That way we could take the new docs and test to the rmw implementation maintainers to ask them to fix it.

we can ask help for rmw implementation maintainers. (`rmw_fastrtps` and `rmw_cyclonedds`)

**Review by ahcorde — APPROVED** [review]
There are some other parts in the tests where we do similar conditions. If we keep the meta issue open to keep track of the other rmw implementations, I'm fine to merge this.

**Comment by ahcorde** [comment]
Pulls: ros2/rclcpp#2636
Gist: https://gist.githubusercontent.com/ahcorde/2cf3487ad132d0176a20e485febb7ca9/raw/4b56e1d716c28091c288491f12979e5dd6bbe8a4/ros2.repos
BUILD args:   --packages-up-to rclcpp --packages-above-and-dependencies rclcpp
TEST args:  --packages-select rclcpp --packages-above rclcpp
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14637
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21920/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16186/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1434/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22642/)

**Pull request merged**